x/mongo/driver/session/client_session.go

```diff
@@ -230,7 +234,12 @@ func NewClientSession(pool *Pool, clientID uuid.UUID, opts ...*ClientOptions) (*
 // SetServer will check out a session from the client session pool.
 func (c *Client) SetServer() error {
 	var err error
-	c.Server, err = c.pool.GetSession()
+	if c.RequireNew {
+		c.Server, err = newServerSession()
```

Comment:
The `session.Server` created here will be returned to the pool when `EndSession` is called. To allow the `CheckedOut` counter to track this new checked-out session, we should call `Pool.createServerSession` here instead.

```suggestion
		c.Server, err = c.pool.createServerSession()
```

Alternatively, if there's concern about pooling these sessions in the Go Driver, we need to find a way to bypass the pool when we end the session.

Will pooling these new sessions in the Go Driver create any issues?

Reply:
Good catch, suggest marking the server as dirty when requesting a new server.